src/science/roughness/cbl_hgtAbove_snow.F90

```diff
@@ -1,47 +1,71 @@
 MODULE cbl_hruff_mod
 
+  !* The procedure in this module evaluates the canopy height
+  !  given the effect of any snow present.  
+
   IMPLICIT NONE
 
   PUBLIC HgtAboveSnow
 
 CONTAINS
-!Computes height above surface given that therre is snow present
-!variable formerly known as rough%hruff
+
+
 subroutine HgtAboveSnow( HeightAboveSnow, mp, z0surf_min, HGT_pft, &
-                         SnowDepth, SnowDensity )
+     SnowDepth, SnowDensity )
+
+  !* This subroutine computes the height of the canopy above ground/snow
+  !  surface when there is snow present. 
+  
   implicit none
 
   !re-decl input args  
-  integer  :: mp
+  integer  :: mp !! Number of land tiles (-)
   !result to return
   real :: HeightAboveSnow(mp) 
+    !! Output. Effective height of canopy, known as `rough%hruff` elsewhere. (m)
   
   !re-decl input args  
-  real :: z0surf_min
-  real :: HGT_pft(mp) 
-  real :: SnowDepth(mp) !snow depth (liquid water ) 
-  real :: SnowDensity(mp) 
+  real :: z0surf_min !! minimum roughness length of the surface (m)
+  real :: HGT_pft(mp) !! height of the canopy without snow (m)
+  real :: SnowDepth(mp) !! amount of snow (mm m(^{-2}\) liquid water) 
```

Comment:
math typo - should be
!! amount of snow (mm m\(^{-2}\) liquid water)

Reply:
aargh - that should be
!! amount of snow (mm m\(^{-2}\) liquid water)

Reply:
okay the comment is not picking up the Markup.  It needs a math start after 'mm m' and a math end after the '2}'

Reply:
Just some help with the Markdown, you need to quote the whole thing with \` or put in a code block to say it is pre-formatted:
`(mm m\(^{-2}\)`

Otherwise `\` is the escape character that allows printing special characters. E.g:

    \# It's not a title if I start with `\`

renders as
\# It's not a title if I start with `\`